corehq/apps/users/tests/test_login_record.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime
+from django.test import SimpleTestCase, TestCase
+
+from ..login_record import LoginRecord, EPOCH
+
+
+class TestLoginRecord_Constructor(SimpleTestCase):
+    def test_creates_valid_object(self):
+        current_time = datetime(2020, 12, 5)
+        record = LoginRecord('test_user', failures=4, last_attempt_date=current_time)
+        self.assertEqual(record.username, 'test_user')
+        self.assertEqual(record.failures, 4)
+        self.assertEqual(record.last_attempt_date, current_time)
+
+    def test_handles_default_arguments(self):
+        record = LoginRecord('test_user')
+        self.assertEqual(record.failures, 0)
+        self.assertEqual(record.last_attempt_date, EPOCH)
+
+
+class TestLoginRecord(TestCase):
```

Comment:
Is this correct, though? I realize redis isn't touching the DBs that django rolls back, but it is running tests against a database. I believe we have some kind of --no-db flag, right? And labelling this as a `SimpleTestCase` would include it in tests not run against a database?

Reply:
It's not a conceptual _database/non-database test_ distinction. Our test runner runs `SimpleTestCase` in an earlier batch of tests with stubbed couch databases objects that raise errors if used and Django disables access via the ORM, so there are guardrails there for the databases that matter.

Reply:
They key distinction is that there is no database setup/teardown for `SimpleTestCase`, unlike `TestCase`. Django takes a lot more time to run `TestCase` tests because it first initializes dbs and runs migrations. We implemented `--reusedb` to attempt to optimize that, but it's still true.